### PR TITLE
[TACHYON-1173] Move report lost file to Lineage Client

### PR DESCRIPTION
**Pull request by calvinjia**
https://tachyon.atlassian.net/browse/TACHYON-1173

### CODE REVIEW

File: bin/tachyon [hunk 1]
```diff
@@ -282,7 +282,7 @@ elif [[ ${COMMAND} == "copyDir" ]]; then
 elif [[ ${COMMAND} == "thriftGen" ]]; then
   rm -rf ${bin}/../common/src/main/java/tachyon/thrift
   for src_file in $(ls ${bin}/../common/src/thrift); do
-    thrift --gen java -out ${bin}/../common/src/main/java/. ${bin}/../common/src/thrift/${src_file}
+    /usr/local/Cellar/thrift/0.9.2/bin/thrift --gen java -out ${bin}/../common/src/main/java/. ${bin}/../common/src/thrift/${src_file}
```
Comment (apc999) [inline bin/tachyon]: do we want this path which looks very specific to mac ?

Comment (calvinjia) [inline bin/tachyon]: whoops

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 2]
```diff
@@ -183,6 +189,9 @@ private TachyonFS(TachyonConf tachyonConf) {
 
     mMasterAddress = NetworkAddressUtils.getConnectAddress(ServiceType.MASTER_RPC, tachyonConf);
     mZookeeperMode = mTachyonConf.getBoolean(Constants.ZOOKEEPER_ENABLED);
+    mLineageEnabled = mTachyonConf.getBoolean(Constants.USER_LINEAGE_ENABLED);
```
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: You can use LineageUtils.isLineageEnabled

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 3]
```diff
@@ -945,7 +954,8 @@ public synchronized boolean rename(long fileId, TachyonURI srcPath, TachyonURI d
    */
   public synchronized void reportLostFile(long fileId) throws IOException {
     try {
-      mFSMasterClient.reportLostFile(fileId);
+      String path = getFile(fileId).getPath();
+      mLineageClient.reportLostFile(path);
```
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: Better exception handling when lineage not enabled?

File: clients/unshaded/src/main/java/tachyon/client/lineage/LineageMasterClient.java [hunk 4]
```diff
@@ -144,4 +144,27 @@ public synchronized void asyncCompleteFile(long fileId) throws IOException {
...
+   * @param path the file path
+   * @throws IOException if an I/O error occurs
+   * @throws TachyonException if a Tachyon error occurs
+   */
+  public synchronized void reportLostFile(String path) throws IOException, TachyonException {
+    int retry = 0;
+    while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
+      connect();
+      try {
+        mClient.reportLostFile(path);
```
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/lineage/LineageMasterClient.java]: Missing return statement.

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java (hunk 2 above)
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: updated, thanks.

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java (hunk 3 above)
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: sure.

File: clients/unshaded/src/main/java/tachyon/client/lineage/LineageMasterClient.java (hunk 4 above)
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/lineage/LineageMasterClient.java]: good catch, thanks!

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**